packages/fxa-auth-server/fxa-oauth-server/test/api.js

```diff
@@ -3620,6 +3621,75 @@ describe('/v1', function() {
           assertSecurityHeaders(res);
         });
     });
+
+    it('should fail if invalid client credentials are provided in authorization header', async () => {
+      const tok = (await newToken()).result;
+      const res = await Server.api.post({
+        url: '/destroy',
+        headers: {
+          authorization: basicAuthHeader(clientId, badSecret),
+        },
+        payload: {
+          token: tok.access_token,
+        },
+      });
+      assert.equal(res.statusCode, 400);
+      assert.equal(res.result.errno, 102);
+    });
+
+    it('should succeed if valid client credentials are provided in authorization header', async () => {
+      const tok = (await newToken()).result;
+      const res = await Server.api.post({
+        url: '/destroy',
+        headers: {
+          authorization: basicAuthHeader(clientId, secret),
+        },
+        payload: {
+          token: tok.access_token,
+        },
+      });
+      assert.equal(res.statusCode, 200);
+    });
+
+    it('should fail if invalid client credentials are provided in request body', async () => {
+      const tok = (await newToken()).result;
+      const res = await Server.api.post({
+        url: '/destroy',
+        payload: {
+          client_id: clientId,
+          client_secret: badSecret,
+          token: tok.access_token,
+        },
+      });
+      assert.equal(res.statusCode, 400);
+      assert.equal(res.result.errno, 102);
+    });
+
+    it('should succeed if valid client credentials are provided in request body', async () => {
```

Comment:
Is there a test that it should fail if both a valid `authorization` header and client credentials are sent?

Reply:
Good idea, added.